src/table/table.component.ts

```diff
@@ -628,13 +628,17 @@ export class Table implements AfterViewInit, OnDestroy {
 	}
 
 	onSelectAll() {
-		this.model.selectAll(true);
-		this.selectAll.emit(this.model);
+		if (this.model.data.length > 1) {
+			this.model.selectAll(true);
```

Comment:
Maybe it makes sense for the model to handle this? If there's no data `selectAll` would still select all the non-existent data right?

Reply:
Yeah, that makes a lot more sense